# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1358711404 18000
#      Sun Jan 20 14:50:04 2013 -0500
# Branch classdef
# Node ID 24ceda35d14688d8a173418cf6d4d4b9bca26521
# Parent  a3d7b927ce471320a925df2748ed9647a5a59ef7
Add classdef tests for method changing the size of a property.

* test/classdef/foo_method_changes_property_size.m: New file.
* test/classdef/module.mk: Add new file.
* test/classdef/test_classdef.m: Add test.

diff --git a/test/classdef/foo_method_changes_property_size.m b/test/classdef/foo_method_changes_property_size.m
new file mode 100644
--- /dev/null
+++ b/test/classdef/foo_method_changes_property_size.m
@@ -0,0 +1,14 @@
+classdef foo_method_changes_property_size
+  properties
+    element;
+  end
+  methods
+    function obj = foo_method_changes_property (n)
+      obj.element = 1:n;
+    end
+    function obj = move_element_to_end (obj, n)
+      obj.element(end+1) = obj.element(n);;
+      obj.element(n) = [];
+    end
+  end
+end
diff --git a/test/classdef/module.mk b/test/classdef/module.mk
--- a/test/classdef/module.mk
+++ b/test/classdef/module.mk
@@ -1,6 +1,7 @@
 classdef_FCN_FILES = \
+  classdef/foo_method_changes_property_size.m \
+  classdef/foo_static_method_constant_property.m \
   classdef/foo_value_class.m \
-  classdef/foo_static_method_constant_property.m \
   classdef/test_classdef.m
 
 FCN_FILES += $(classdef_FCN_FILES)
diff --git a/test/classdef/test_classdef.m b/test/classdef/test_classdef.m
--- a/test/classdef/test_classdef.m
+++ b/test/classdef/test_classdef.m
@@ -62,8 +62,13 @@
 %!assert (foo_static_method_constant_property().pie, pi);
 %!error <property `frequency' is not constant> foo_static_method_constant_property.frequency
 %!error <method `cosine' is not static> foo_static_method_constant_property.cosine
 %!test
 %! obj = foo_static_method_constant_property;
 %! obj.frequency = 10;
 %! assert (obj.cosine (0.1), cos (2 * pi * 10 * 0.1), eps ())
 %! assert (obj.sine (0.1), sin (2 * pi * 10 * 0.1), eps ())
+
+%!xtest
+%! obj = foo_method_changes_property (3);
+%! obj = obj.move_element_to_end (2);
+%! assert (obj.element, [1 3 2])
